modules/genflow-api/src/main/java/com/reprezen/genflow/api/util/FileUtils.java

```diff
@@ -80,8 +87,15 @@ private FileUtils() {
 		// resolve "." and ".." compoents. We prefix with "/" and remove afterward, in
 		// order to preventing resolving relative to current working directory in
 		// filesystem (e.g. "../foo" would do that)
-		source = new File("/" + source).getCanonicalPath().substring(1);
-		if (source.startsWith("../")) {
+		source = new File("/" + source).getCanonicalPath().replaceAll(Pattern.quote(File.separator), "/");
+		if (source.contains("/")) {
+			// skip initial slash and (on windows) device letter) for matching with jar file
+			// entries
+			source = source.substring(source.indexOf('/') + 1);
+		}
+		if (source.startsWith(".." + "/"))
```

Comment:
Not sure why the string "../" needed to be expressed as a concatentation. 

Also, the preceding conditional branch trims everything up to the first slash, inclusive. If the slash-normalized `source` path started with "../", then that would already have been trimmed. So this code branch gets called _only if_ the slash-normalized `source` had another "../" sequence _after_ the first one. For example, `"../../files/foo"`.

Is that correct?

Reply:
@tedepstein The concatenation only because I originally coded it using the `File.separator` constant, but then realized I would need it to be slashes eventually anyway, so I just did that replacement above, and now this could be a single constant.

This is needed because the resource paths declared in gentemplates sometime include initial ".." segments, but they're resolved relative to the package (as a path) of the gentemplate class that's executing. In some cases, it's useful to locate the classes in a different package, e.g. when two variants of the same GenTemplate use precisely the same content for a static resource. That happens with diagram gentemplate, for example. So the constructed file path will have `..` internal segments in cases like that - the gentarget output directpry path + package path on the left, and static resource path on the right, and we use File.getCanonicalPath to resolve those segments.